db/fdb_fend.c

```diff
@@ -692,6 +693,7 @@ static int _table_exists(fdb_t *fdb, const char *table_name,
  * They really belong to the fdb, lets properly link them now
  *
  * Returns -1 for ENOMEM
+ * Returns -2 if cannot find stat_name
```

Comment:
I'd avoid using hardcoded error codes (-1 is generic enough, though). It also appears that we do not handle the error codes that differently. Can we move the logmsg calls for printing errors into this function, and return -1 for both error cases? For instance,

```
if (!stat_tbl) {
    logmsg(LOGMSG_ERROR, "%s: OOM stat1 for %p\n", __func__, tbl);
    return -1;
}

// ...

if (!stat_ent) {
    logmsg(LOGMSG_ERROR, "%s: Cannot find stat1 for %p\n", __func__, tbl);
    return -1;
}

```

Reply:
That makes sense, will do!